dev/WindowsAppRuntime_BootstrapDLL/MddBootstrapActivity.h

```diff
@@ -73,17 +79,45 @@ namespace WindowsAppRuntime::MddBootstrap::Activity
             return m_lastFailure;
         }
 
-        void SetMddBootStrapAPI(MddBootStrapAPI mddBootStrapAPI)
+        void SetMddBootstrapAPI(MddBootstrapAPI mddBootstrapAPI)
+        {
+            m_mddBootstrapAPI = mddBootstrapAPI;
+        }
+
+        void SetInitializeData()
         {
-            m_mddBootStrapAPI = mddBootStrapAPI;
+            ++g_initializationCount;
         }
 
-        void SetDDLMFindMethodUsed(DDLMFindMethod ddlmFindMethodUsed)
+        // Returns a boolean to indicate to the caller if MddBootstrap should be shutdown
+        bool ShouldShutdownNow()
         {
-            m_ddlmFindMethodUsed = ddlmFindMethodUsed;
+            if (g_initializationCount > 1)
+            {
+                // Multiply initialized. Just decrement our count
+                --g_initializationCount;
+                return false;
+            }
+            else if (g_initializationCount == 0)
+            {
+                // Not initialized. Nothing to do
+                return false;
+            }
+            return true;
         }
 
         void SetLastFailure(const wil::FailureInfo& failure);
+
+        void SetShutdownData()
+        {
+            g_initializationPackageFullName = {};
+            --g_initializationCount;
```

Comment:
This always decrements, but what if count was already zero? Does that caller trap that?

This always clears the pkgfullname, but what if count was >1? Wouldn't we lose the value?

And why do we have both `SetShutdownData()` and `ShouldShutdownNow()`?

Confusing.

Reply:
Agree. Removed SetShutdownData.